diff --git a/mm/init-mm.c b/mm/init-mm.c
index 1d29cdf..fc8201e 100644
--- a/mm/init-mm.c
+++ b/mm/init-mm.c
@@ -4,6 +4,7 @@
 #include <linux/spinlock.h>
 #include <linux/list.h>
 #include <linux/cpumask.h>
+#include <linux/module.h>
 
 #include <asm/atomic.h>
 #include <asm/pgtable.h>
@@ -24,3 +25,4 @@ struct mm_struct init_mm = {
 	.cpu_vm_mask	= CPU_MASK_ALL,
 	INIT_MM_CONTEXT(init_mm)
 };
+EXPORT_SYMBOL(init_mm);
